tq/basic_upload.go

```diff
@@ -94,6 +102,15 @@ func (a *basicUploadAdapter) DoTransfer(ctx interface{}, t *Transfer, cb Progres
 
 	res, err := a.doHTTP(t, req)
 	if err != nil {
+		// We're about to return a retriable error, meaning that this
+		// transfer will either be retried, or it will fail.
+		//
+		// Either way, let's decrement the number of bytes that we've
+		// read _so far_, so that the next iteration doesn't re-transfer
+		// those bytes, according to the progress meter.
+		offset := fcount.N()
+		ccb(t.Size, offset, -int(offset))
```

Comment:
If this was using the progress package's `*bodyWithCallback`, it could just be a single method like `ResetProgress()`.